src/test/java/seedu/commands/BudgetCommandTest.java

```diff
@@ -93,4 +103,81 @@ public void execute_tooFewInputs_exceptionThrown() {
+                System.lineSeparator(), outContent.toString());
+    }
+
+    @Test
+    public void execute_monthlyLessThanDailyBudget_monthlyLessThanDailyMessage() {
+        commandInputForTest("budget monthly 1", incomes, spendings);
+        assertEquals(TAB + "Daily budget should not be larger than monthly budget! " + ENTER_BUDGET_MESSAGE +
+                System.lineSeparator(), outContent.toString());
+    }
+
+    @Test
+    public void execute_yearlyMoreThanMonthlyBudget_yearlyMoreThanMonthlyMessage() {
+        commandInputForTest("budget yearly 100", incomes, spendings);
+        assertEquals(TAB + "Monthly budget should not be larger than yearly budget! " + ENTER_BUDGET_MESSAGE +
+                System.lineSeparator() ,outContent.toString());
+    }
+
+    @Test
+    public void initialiseBudget_dailyBudgetLessThanZero_invalidBudgetMessage() {
+        Ui.userInputForTest("-1" + System.lineSeparator() + "100" + System.lineSeparator() + "1000" +
+                System.lineSeparator() + "10000");
+        BudgetCommand.initialiseBudget(spendings);
+        assertEquals(TAB + DAILY_BUDGET_QUESTION + System.lineSeparator() + TAB + SEPARATOR +
+                System.lineSeparator() + TAB + INVALID_AMOUNT + ENTER_BUDGET_MESSAGE + System.lineSeparator() + TAB +
+                MONTHLY_BUDGET_MESSAGE + System.lineSeparator() + TAB + SEPARATOR + System.lineSeparator() + TAB +
+                YEARLY_BUDGET_MESSAGE + System.lineSeparator() + TAB + SEPARATOR + System.lineSeparator(),
+                outContent.toString());
+    }
+
+
+    @Test
+    public void initialiseBudget_budgetLessThanZero_invalidBudgetMessage() {
+        Ui.userInputForTest("-1" + System.lineSeparator() + "100" + System.lineSeparator() + "1000" +
+                System.lineSeparator() + "10000");
+        BudgetCommand.initialiseBudget(spendings);
+        assertEquals(TAB + DAILY_BUDGET_QUESTION + System.lineSeparator() + TAB + SEPARATOR +
+                System.lineSeparator() + TAB + INVALID_AMOUNT + ENTER_BUDGET_MESSAGE + System.lineSeparator() + TAB +
+                MONTHLY_BUDGET_MESSAGE + System.lineSeparator() + TAB + SEPARATOR + System.lineSeparator() + TAB +
+                YEARLY_BUDGET_MESSAGE + System.lineSeparator() + TAB + SEPARATOR + System.lineSeparator(),
+                outContent.toString());
+    }
+
+    @Test
+    public void initialiseBudget_budgetMoreThanLimit_invalidBudgetMessage() {
+        Ui.userInputForTest("100" + System.lineSeparator() + "10000000000" + System.lineSeparator() + "1000" +
+                System.lineSeparator() + "10000");
+        BudgetCommand.initialiseBudget(spendings);
+        assertEquals(TAB + DAILY_BUDGET_QUESTION + System.lineSeparator() + TAB + SEPARATOR +
+                System.lineSeparator() + TAB + MONTHLY_BUDGET_MESSAGE + System.lineSeparator() + TAB + SEPARATOR +
+                System.lineSeparator() + TAB + "Amount must be lesser than 1.0E8Please enter a valid budget:" +
```

Comment:
Personally I think 1.0E8 looks a bit weird, also is there a space between 1.08E and Please?

Reply:
not sure how to change as it is a double

Reply:
For add and edit, I wrote `100 million` directly in the error string. It may not be a good practice but I think it is a better way to format for now.

Reply:
Maybe can create a constant just for the string value of the max list amount

Reply:
okie done